cms/server/contest/handlers/base.py

```diff
@@ -66,26 +67,23 @@
 NOTIFICATION_SUCCESS = "success"
 
 
-def check_ip(ip, whitelist):
-    """Return if client IP belongs to one of the accepted subnets.
+def check_ip(address, networks):
+    """Return if client IP belongs to one of the accepted networks.
 
-    ip (string): IP address to verify.
-    whitelist (string): IP addresses or subnets to check against (separated
-        by a comma).
+    address (string): IP address to verify.
+    networks ([ipaddress.IPv4Network|ipaddress.IPv6Network]): IP
+        networks (addresses w/ subnets) to check against.
 
-    return (bool): whether client is equal to one of the IPs in the whitelist
-        or client belongs to one of the subnets in the whitelist.
+    return (bool): whether the address belongs to one of the networks.
 
     """
-    for wanted in whitelist.split(","):
-        wanted, sep, subnet = wanted.partition('/')
+    try:
+        address = ipaddress.ip_address(address)
+    except ValueError:
+        return False
```

Comment:
This should never happen because of DB validation, right? If that's the case, wouldn't it be better to throw an error?